### PR TITLE
UIIN-2753: Add the ability to update ownership of holdings

### PR DESCRIPTION
**Pull request by OleksandrHladchenko1**
## Purpose

 * Add ability to move holdings from one FOLIO tenant to another.

## Approach

* Added `updateOwnership` function that sends POST request in order to update ownership of holdings.
* Added `omitCurrentAndCentralTenants` function in order to filter active member tenant and central one (we need this filtered list to pass it to location lookup).
* Display "Update ownership" button when:
 1) Instance is shared.
 AND
 2) User has "Inventory: Update ownership" permission.
 AND
 3) When user has multiple affiliations (except active member tenant and central one).

* When user clicks "Update ownership" button the location lookup is shown.
* After choosing the location confirmation modal is shown.
* When "Confirm" button is clicked POST request is sent.
* When request is performed successfully a green message is shown.
* When request is failed a "Server communication problem" message is shown.

## Refs

[UIIN-2753](https://folio-org.atlassian.net/browse/UIIN-2753)

## Screenshots

https://github.com/user-attachments/assets/bd32d157-0ce5-4869-b423-a44a3830eb66

### CODE REVIEW

File: src/ViewHoldingsRecord.js [hunk 1]
```diff
@@ -417,6 +498,19 @@ class ViewHoldingsRecord extends React.Component {
...
+          <Button
+            id="update-ownership"
+            onClick={this.openLocationLookup}
+            buttonStyle="dropdownItem"
+            data-testid="update-ownership-btn"
+          >
+            <Icon icon="profile">
+              <FormattedMessage id="ui-inventory.updateOwnership" />
+            </Icon>
+          </Button>
```
Comment (mariia-aloshyna) [inline src/ViewHoldingsRecord.js:512]: Use `ActionItem` component, please

File: src/ViewHoldingsRecord.test.js [hunk 2]
```diff
@@ -397,4 +412,140 @@ describe('ViewHoldingsRecord actions', () => {
       });
     });
   });
+
+describe('Update ownership action item', () => {
```
Comment (mariia-aloshyna) [inline src/ViewHoldingsRecord.test.js:416]: fix eslint errors, please

File: src/ViewHoldingsRecord.test.js [hunk 3]
```diff
@@ -397,4 +412,140 @@ describe('ViewHoldingsRecord actions', () => {
...
+        },
+      }),
+    };
+
+    describe('when instance is shared', () => {
+      it('should be rendered', async () => {
+        renderViewHoldingsRecord({ stripes, isInstanceShared: true });
+
+        const updateOwnershipBtn = await screen.findByTestId('update-ownership-btn');
+        expect(updateOwnershipBtn).toBeInTheDocument();
```
Comment (mariia-aloshyna) [inline src/ViewHoldingsRecord.test.js:448]: Can getByRole() be used here?
```suggestion
        expect(screen.getByRole('button', { name: 'Update ownership' })).toBeInTheDocument();;
```

File: src/ViewHoldingsRecord.test.js [hunk 4]
```diff
@@ -397,4 +412,140 @@ describe('ViewHoldingsRecord actions', () => {
...
+            targetTenantId,
+            toInstanceId: defaultProps.resources.holdingsRecords.records[0].instanceId,
+            targetLocationId,
+          },
+          { tenant: stripes.okapi.tenant }
+        );
+      });
+    });
+
+    describe('when an error was occured updating ownership', () => {
```
Comment (mariia-aloshyna) [inline src/ViewHoldingsRecord.test.js:515]:
```suggestion
    describe('when an error was occurred during updating ownership', () => {
```

File: src/utils.js [hunk 5]
```diff
@@ -1055,3 +1056,21 @@ export const flattenCentralTenantPermissions = (centralTenantPermissions) => {
...
+export const omitCurrentAndCentralTenants = (stripes) => {
+  const tenants = stripes?.user?.user?.tenants;
+
+  const currentTenantId = stripes.okapi.tenant;
+  const centralTenantId = stripes.user?.user?.consortium?.centralTenantId;
+
+  return tenants?.filter(tenant => tenant.id !== currentTenantId && tenant.id !== centralTenantId);
+};
+
+export const updateOwnership = (body, okapi) => {
```
Comment (mariia-aloshyna) [inline src/utils.js:1069]: can react query mutator be used here?

**Review by mariia-aloshyna — APPROVED** [review]

**Review by SKarolFolio — APPROVED** [review]

**Pull request merged**